### PR TITLE
WIP [17.07] fix tests

### PR DESCRIPTION
**Pull request by andrewhsu**
Fixes the following tests:

- DockerSuite.TestRmiContainerImageNotFound (caused by docker/cli#160)
- DockerSuite.TestRmiImageIDForceWithRunningContainersAndMultipleTags (caused by docker/cli#160)
- DockerSuite.TestRunWithNanoCPUs (caused by docker/cli#202)
- DockerSuite.TestUpdateWithNanoCPUs (caused by docker/cli#202)
- DockerSwarmSuite.TestServiceLogs (caused by docker/cli#121)
- DockerSwarmSuite.TestSwarmNetworkPluginV2 (caused by docker/cli#121)
- DockerSwarmSuite.TestSwarmServicePsMultipleServiceIDs (caused by docker/cli#72)

Punting on:
- DockerSuite.TestRunAttachDetachFromInvalidFlag (investigating in docker/cli#380)

Test verified by compiling cli from docker-ce 17.07 branch and using that when running test-integration-cli test from engine component of docker-ce 17.07 branch, e.g.
```
$ make BIND_DIR=. \
  DOCKER_CLI_PATH=/path/to/cli/build/docker-linux-amd64 \
  TESTFLAGS='-check.f DockerSuite.TestRmiContainerImageNotFound' \
  test-integration-cli
PASS: docker_cli_rmi_test.go:227: DockerSuite.TestRmiContainerImageNotFound	2.680s
OK: 1 passed
```

### CODE REVIEW

**Comment by seemethere** [comment]
Re-kicking the windows test, failed with a weird error:

```
00:01:17.208 re-exec error: exit status 1: output: time="2017-07-26T00:00:16Z" level=error msg="hcsshim::ImportLayer failed in Win32: The system cannot find the path specified. (0x3) layerId=\\\\?\\D:\\control\\windowsfilter\\5d0ffc3f5a5700b571e8688e653501f634989926441ac1e52dc7627125283cfb flavour=1 folder=d:\\temp\\hcs591301103" 
00:01:17.209 hcsshim::ImportLayer failed in Win32: The system cannot find the path specified. (0x3) layerId=\\?\D:\control\windowsfilter\5d0ffc3f5a5700b571e8688e653501f634989926441ac1e52dc7627125283cfb flavour=1 folder=d:\temp\hcs591301103
00:01:17.209 
00:01:17.209 
00:01:17.210 ERROR: Failed 'ERROR: Failed to build image from Dockerfile.windows' at 07/26/2017 00:00:27
```

Possibly related to this:

```
00:00:53.153 INFO: Container count on control daemon to delete is 1
00:00:58.514 Error response from daemon: driver "windowsfilter" failed to remove root filesystem for a7fe2d1a2db2be19dd76ec62002888cb536e430e6d2fde72ca5c012f20aea8c3: rename D:\control\windowsfilter\a7fe2d1a2db2be19dd76ec62002888cb536e430e6d2fde72ca5c012f20aea8c3 D:\control\windowsfilter\a7fe2d1a2db2be19dd76ec62002888cb536e430e6d2fde72ca5c012f20aea8c3-removing: Access is denied.
```

File: components/engine/integration-cli/docker_cli_rmi_test.go [hunk 1]
```diff
@@ -150,8 +150,8 @@ func (s *DockerSuite) TestRmiImageIDForceWithRunningContainersAndMultipleTags(c
 	dockerCmd(c, "tag", imgID, newTag)
 	runSleepingContainerInImage(c, imgID)
 
-	out, _, err := dockerCmdWithError("rmi", "-f", imgID)
-	// rmi -f should not delete image with running containers
+	out, _, err := dockerCmdWithError("rmi", imgID)
+	// rmi should not delete image with running containers
```
Comment (dnephin) [inline components/engine/integration-cli/docker_cli_rmi_test.go]: If we're interested in preserving the intent  of this test I think the correct change is to check that this text is in stderr, but there is no error. It should continue to use `-f`

File: components/engine/integration-cli/docker_cli_rmi_test.go [hunk 2]
```diff
@@ -243,7 +243,7 @@ func (s *DockerSuite) TestRmiContainerImageNotFound(c *check.C) {
 	dockerCmd(c, "rmi", "-f", imageIds[1])
 
 	// Try to remove the image of the running container and see if it fails as expected.
-	out, _, err := dockerCmdWithError("rmi", "-f", imageIds[0])
+	out, _, err := dockerCmdWithError("rmi", imageIds[0])
```
Comment (dnephin) [inline components/engine/integration-cli/docker_cli_rmi_test.go]: same here

**Comment by vieux** [comment]
looks like @tiborvass's fix works

**Comment by dnephin** [comment]
I think the test fixes for `rmi -f` are wrong. I think we should revert https://github.com/docker/cli/pull/160 in this branch instead.  We need to revisit the `-f` behaviour.

I opened https://github.com/docker/cli/issues/394 to track the problem. I can prepare a revert commit of you agree.

**Comment by tiborvass** [comment]
LGTM

**Comment by andrewhsu** [comment]
no longer needed because 17.07 is eol

**Pull request closed without merging**